core/src/main/java/hudson/model/Queue.java

```diff
@@ -3036,7 +3037,9 @@ public static void init(Jenkins h) {
         @VisibleForTesting
         /*package*/ static /*final*/ int DELAY_SECONDS = SystemProperties.getInteger("hudson.model.Queue.Saver.DELAY_SECONDS", 60);
 
-        private volatile Future<?> nextSave;
+        private final Object lock = new Object();
```

Comment:
🐜  Could just synchronize on `this` or do 'synchronized' on methods, but ::shrug:: it's safe now.

Reply:
I have a preference for dedicated private monitor.